File: ProjC1/src/projc1/Game/Player.java
package projc1.Game;
import java.awt.Graphics2D;
import java.net.URL;
import java.awt.Image;
import java.awt.Toolkit;
class Player {
    private Position position; // COMPOSICIÓN: Player tiene Position
    private static final int SIZE = 30;
    private static final int SPEED = 5;
    private Image playerImage;

    public Player(int startX, int startY) {
        this.position = new Position(startX, startY); // COMPOSICIÓN
        playerImage = Toolkit.getDefaultToolkit().getImage(getClass().getResource("/res/cow.png"));

    }
    
    public void moveLeft(int maxWidth) {
        if (position.getX() > 0) {
            position.move(-SPEED, 0);
        }
    }
    
    public void moveRight(int maxWidth) {
        if (position.getX() < maxWidth - SIZE) {
            position.move(SPEED, 0);
        }
    }
    
    public void moveUp() {
        if (position.getY() > 0) {
            position.move(0, -SPEED);
        }
    }
    
    public void moveDown(int maxHeight) {
        if (position.getY() < maxHeight - SIZE) {
            position.move(0, SPEED);
        }
    }
    
    public void reset() {
        position.setPosition(50, 50);
    }
    
    // ASOCIACIÓN: Usa RenderEngine para dibujarse
    public void draw(java.awt.Graphics2D g2d, RenderEngine renderer) {
        int x = position.getX();
        int y = position.getY();
        g2d.drawImage(playerImage, x, y, SIZE, SIZE, null);

    }

    public void setPlayerImage(String resourcePath) {
        if (resourcePath.startsWith("src/")) {
            resourcePath = resourcePath.substring(3); 
        }
        URL url = getClass().getResource(resourcePath);
        if (url == null) {
            System.err.println("ERROR: No se encontró la imagen: " + resourcePath);
            return;
        }

        System.out.println("Imagen cargada correctamente desde: " + url.toString());
        playerImage = Toolkit.getDefaultToolkit().getImage(url);
    }
    
    public Position getPosition() { return position; }
    public int getSize() { return SIZE; }
}

File: ProcC1_pyver/game/entities/player.py
from PIL import Image, ImageTk
import tkinter as tk
from game.components.position import Position
class Player:
    SIZE = 30
    SPEED = 5
    
    def __init__(self, start_x, start_y):
        self.position = Position(start_x, start_y)  # Composicion
        self.player_image = None
        self.image_tk = None
    
    def move_left(self, max_width):
        if self.position.get_x() > 0:
            self.position.move(-self.SPEED, 0)
    
    def move_right(self, max_width):
        if self.position.get_x() < max_width - self.SIZE:
            self.position.move(self.SPEED, 0)
    
    def move_up(self):
        if self.position.get_y() > 0:
            self.position.move(0, -self.SPEED)
    
    def move_down(self, max_height):
        if self.position.get_y() < max_height - self.SIZE:
            self.position.move(0, self.SPEED)
    
    def reset(self):
        self.position.set_position(50, 50)
    
    def draw(self, canvas, renderer):
        x = self.position.get_x()
        y = self.position.get_y()
        
        if self.image_tk:
            canvas.create_image(x, y, image=self.image_tk, anchor=tk.NW)
        else:
            renderer.draw_oval(canvas, x, y, self.SIZE, self.SIZE, "blue")
    
    def set_player_image(self, image_path):
        try:
            img = Image.open(image_path)
            img = img.resize((self.SIZE, self.SIZE), Image.LANCZOS)
            self.image_tk = ImageTk.PhotoImage(img)
        except Exception as e:
            print(f"Error loading player image: {e}")
            self.image_tk = None
    
    def get_position(self): return self.position
    def get_size(self): return self.SIZE